### PR TITLE
[R-package] Remove paste0 ambiguity for paths

### PR DESCRIPTION
**Pull request by Laurae2**
Fixes #607 `paste0` ambiguity.

### CODE REVIEW

**Comment by Laurae2** [comment]
(to not merge yet)

File: R-package/src/install.libs.R [hunk 1]
```diff
@@ -62,7 +62,7 @@ if (!use_precompile) {
 } else {
 
   # Has precompiled package
-  lib_folder <- paste0(R_PACKAGE_SOURCE, "/../")
+  lib_folder <- file.path(R_PACKAGE_SOURCE, "../", fsep "/")
```
Comment (guolinke) [inline R-package/src/install.libs.R]: missing ```=``` ?

Comment (Laurae2) [inline R-package/src/install.libs.R]: yes I noticed it

**Comment by Laurae2** [comment]
Seems to not pass installation when using `file.path`, which is really weird..

**Comment by Laurae2** [comment]
This is what I get with `file.path`:

```r
> devtools::install_github("Laurae2/LightGBM/R-package@patch-2")
Downloading GitHub repo Laurae2/LightGBM@patch-2
from URL https://api.github.com/repos/Laurae2/LightGBM/zipball/patch-2
Installing lightgbm
"C:/PROGRA~1/R/R-34~1.0/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/tmp/RtmpMdQoFq/devtoolsb5881cc95041/Laurae2-LightGBM-404c89c/R-package" --library="C:/Program Files/R/R-3.4.0/library" --install-tests 

* installing *source* package 'lightgbm' ...
** libs
installing via 'install.libs.R' to C:/Program Files/R/R-3.4.0/library/lightgbm
CMake Error: The source directory "C:/tmp/RtmpMdQoFq/devtoolsb5881cc95041/Laurae2-LightGBM-404c89c/R-package" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
Warning: running command 'cmake -DCMAKE_GENERATOR_PLATFORM=x64  ..' had status 1
Error: could not load cache
Warning: running command 'cmake --build . --target _lightgbm  --config Release' had status 1
Error in eval(ei, envir) : Cannot find lib_lightgbm.dll
* removing 'C:/Program Files/R/R-3.4.0/library/lightgbm'
* restoring previous 'C:/Program Files/R/R-3.4.0/library/lightgbm'
Error: Command failed (1)
In addition: Warning message:
GitHub repo contains submodules, may not function as expected!
```

**Comment by guolinke** [comment]
@Laurae2 
from this:
```
CMake Error: The source directory "C:/tmp/RtmpMdQoFq/devtoolsb5881cc95041/Laurae2-LightGBM-404c89c/R-package" does not appear to contain CMakeLists.txt.
```
it seems the working directory isn't in the ```R-package/src/build```.

**Comment by guolinke** [comment]
@Laurae2 
you may need to change this as well.
https://github.com/Laurae2/LightGBM/blob/404c89c5d68af8c237fbf5ff09e45e9f8ac3afe2/R-package/src/install.libs.R#L32

**Comment by Laurae2** [comment]
@guolinke I fixed installation for devtools, but it is not working when using precompiled DLL now. to not merge yet

**Comment by Laurae2** [comment]
@guolinke I think you can merge, unless you have specific things to add/comment/change.

Tested under:

* `devtools::install_github` (Visual Studio installation)
* `lgbdl::lgb.dl` with precompiled DLL
* `lgbdl::lgb.dl` without precompiled DLL (MinGW / Visual Studio installation)

**Pull request merged**